Added support for YAML as an output format

VaibhavMalik4187 opened the pull request:
This commit adds the support for using yaml as an output format to store the reports.

For Example: kubescape scan image nginx -f yaml

The report will be stored in the "report.yaml" file.

Fixes: https://github.com/kubescape/kubescape/issues/1395

 
## Screenshot
![image](https://github.com/kubescape/kubescape/assets/34866732/30a0a402-d654-48e6-8e71-286e30729f22)

<!--
> Any additional information that may be useful for reviewers to know 
-->

<!--
## How to Test

> Please provide instructions on how to test the changes made in this pull request
-->

<!--
## Examples/Screenshots

> Here you add related screenshots 
-->

<!-- 
## Related issues/PRs:

Here you add related issues and PRs.
If this resolved an issue, write "Resolved #<issue number>

e.g. If this PR resolves issues 1 and 2, it should look as follows:
* Resolved #1
* Resolved #2
-->

<!--
## Checklist before requesting a review

put an [x] in the box to get it checked 

- [X] My code follows the style guidelines of this project
- [X] I have commented on my code, particularly in hard-to-understand areas
- [X] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests.
- [ ] New and existing unit tests pass locally with my changes

-->

dwertent commented:
@craigbox this YAML output will not be aligned with the Kubescape-Operator YAML format, is this ok by you?

craigbox commented:
update the issue to point out the differences and let's see if it can be
fixed

On Sun, Oct 1, 2023 at 8:48 PM David Wertenteil ***@***.***>
wrote:

> @craigbox <https://github.com/craigbox> this YAML output will not be
> aligned with the Kubescape-Operator YAML format, is this ok by you?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/kubescape/kubescape/pull/1401#issuecomment-1741993727>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AABALHTBGKKN72TBDMIC3QLX5EN3ZANCNFSM6AAAAAA5NSJ7V4>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

dwertent commented:
@VaibhavMalik4187 Thank you for contributing :)
It seems like the issue was not clear enough, so I added [here](https://github.com/kubescape/kubescape/issues/1395#issuecomment-1742019887) some more details of how we visioned the solution.
Please let me know if it is more clear.

VaibhavMalik4187 commented:
> @VaibhavMalik4187 Thank you for contributing :) It seems like the issue was not clear enough, so I added [here](https://github.com/kubescape/kubescape/issues/1395#issuecomment-1742019887) some more details of how we visioned the solution. Please let me know if it is more clear.

I tried looking for instances of yaml in the Kubescape-Operator codebase using the following command:
`git grep -r yaml "*.go"`

Didn't get a lot of data but I noticed that `mainhandler/cronjobhandler.go` is using "sigs.k8s.io/yaml" and I was using "gopkg.in/yaml.v3". I've updated the code. Now the yaml files are indented by 2 spaces instead of 4.

VaibhavMalik4187 commented:
@dwertent @craigbox ...

matthyx commented:
just converting the JSON to a yaml doesn't bring any value to the user
what would make sense would be to generate a real CRD as yaml
